metagraph/src/cli/query.hpp

```diff
@@ -28,24 +29,43 @@ class Config;
 
 using StringGenerator = std::function<void(std::function<void(const std::string &)>)>;
 
+/**
+ * Construct a query graph
+ * @param anno_graph the input annotated de Bruijn graph
+ * @param call_sequences generate sequences to be queried against anno_graph
+ * @param discovery_fraction the minimum fraction at which a generated sequence
+ *                           must be present in anno_graph to be considered
+ * @param num_threads number of threads to use
+ * @param canonical if true, the returned query graph is a canonical graph
+ * @param sub_k also include k-mers whose suffixes of length sub_k match sub_k-mers
+ *              in the generated sequences. This is only applicable for DBGSuccinct
+ * @param max_hull_forks when adding a hull to the query graph from anno_graph,
+ *                       restrict traversal to at most max_hull_forks forking nodes
+ * @param max_hull_depth nodes added to the hull must be within the max_hull_depth
+ *                       neighbourhood of k-mers in the generated sequences
+ * @param max_hull_depth_per_seq_char restrict the traversal around a generated
+ *                                    sequence's k-mers to its
+ *                                    len(seq) * max_hull_depth_per_seq_char
+ *                                    neighbourhood.
+ */
```

Comment:
The idea here is that I wanted to have the traversal depth/distance be a function of the length of the input sequences, though this is worst-case behaviour since the traversal stops if it finds a matching k-mer.

But if the user passes in a very long query sequence, I still wanted to have `max-hull-depth` as a hard cutoff on the traversal.